packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/boolean_state_var.test.rs

```diff
@@ -0,0 +1,423 @@
+
+    let dependencies_created_for_data_queries =
+        vec![DependenciesCreatedForDataQuery(vec![child_dependency])];
+
+    state_var.save_dependencies(&dependencies_created_for_data_queries);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 3: check if get the correct calculated value
+    ////////////////////////////////////////////////////////////////
+    state_var.calculate_and_mark_fresh();
+
+    // we expect a value of true and came_from_default to be false,
+    // since that the variable is coming from the child dependency, which we initialized to true.
+    assert_eq!(*state_var.get(), true);
+    assert_eq!(state_var.came_from_default(), false);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 4: invert to make the value false
+    ////////////////////////////////////////////////////////////////
+
+    // on the state variable view, record that we request the value be false
+    state_var_view.queue_update(false);
+
+    let invert_result = state_var.invert(false).unwrap();
+
+    // we should get a request informing core that we need to change the child variable
+    assert_eq!(
+        invert_result,
+        vec![DependencyValueUpdateRequest {
+            // data_query_idx is 0 because child is the first data query
+            data_query_idx: 0,
+            dependency_idx: 0
+        }]
+    );
+    // the child variable has recorded that it has been requested to be false
+    assert_eq!(child_var.get_requested_value().to_string(), "false");
+
+    ////////////////////////////////////////////////////////////////
+    // Step 5: make the changes to actually make the value false
+    ////////////////////////////////////////////////////////////////
+
+    child_var.set_value(String::from("false"));
+
+    state_var.calculate_and_mark_fresh();
+
+    // now the value should be false
+    assert_eq!(*state_var.get(), false);
+}
+
+/// Testing the case of boolean state variable requesting children
+/// and receiving a single string child dependency that is an empty string.
+/// For example, if we have a `<boolean><text></text></boolean>`,
+/// its `value` state variable would follow this pattern.
+#[test]
+fn empty_child_string_is_false() {
+    let mut state_var = BooleanStateVar::new_from_children(false).into_state_var();
+
+    // need to return data queries since side effect is saving the required data
+    state_var.return_data_queries();
+
+    let (child_dependency, _child_var) = create_state_var_dependency(String::from(""), false);
+
+    let dependencies_created_for_data_queries =
+        vec![DependenciesCreatedForDataQuery(vec![child_dependency])];
+
+    state_var.save_dependencies(&dependencies_created_for_data_queries);
+
+    state_var.calculate_and_mark_fresh();
+
+    assert_eq!(*state_var.get(), false);
+}
+
+/// Testing the case of boolean state variable is case insensitive when has string children.
+#[test]
+fn case_insensitive() {
```

Comment:
-> `can_convert_string_to_boolean_in_case_insensitive_way()`

Tests are a place where you have have long function names ;-). More specifically, it's helpful when you see which tests failed, if you know what might be going wrong, rather than having to refer to the source of the test.

Reply:
tried to do that!